docs/playbooks/wind-river-cloud-platform-deployment-manager.yaml

```diff
@@ -225,110 +183,59 @@
+                 Expected 'applied' state but the app is in 
+                 '{{ app_apply_status.stdout }}' state."
+         when: not ("applied" in app_apply_status.stdout)
 
-    # Remove v1 webhook and webhook service
-    - block:
-      - name: Search webhook configurations for v1
-        shell: |
-          kubectl get ValidatingWebhookConfigurations validating-webhook-configuration --no-headers | awk 'NR == 1 { print $1 }'
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-        ignore_errors: yes
-        register: validation_webhook
-
-      - name: Query for validating webhook configuration
-        shell: |
-          grep -iq 'validating-webhook-configuration' <<< {{ validation_webhook.stdout_lines }}
-        ignore_errors: yes
-        register: webhook_config_exists
-
-      - name: Delete validating webhook configuration
-        shell: |
-          kubectl delete ValidatingWebhookConfigurations validating-webhook-configuration
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-        when: webhook_config_exists.rc == 0
-
-      - name: Search webhook service for v1
-        shell: |
-          kubectl get svc webhook-server-service -n platform-deployment-manager --no-headers | awk 'NR == 1 { print $1 }'
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-        register: webhook_service
-
-      - name: Query for webhook-server-service
-        shell: |
-          grep -iq 'webhook-server-service' <<< {{ webhook_service.stdout_lines }}
-        ignore_errors: yes
-        register: webhook_service_exists
-
-      - name: Delete webhook-server-service
-        shell: |
-          kubectl delete svc webhook-server-service -n platform-deployment-manager
-        environment:
-          KUBECONFIG: "/etc/kubernetes/admin.conf"
-        when: webhook_service_exists.rc == 0
-
-      when: helmv3_installed and helmv3_dm_exists.rc == 0
-
-    - name: Install Deployment Manager
-      shell: KUBECONFIG=/etc/kubernetes/admin.conf /usr/sbin/helm upgrade --install deployment-manager {% if helm_chart_overrides is defined %}--values {{ helm_chart_overrides }}{% endif %} {{ manager_chart }}
-      when: helmv3_installed
-
-    # Install or Upgrade DM-Monitor
-    - name: "Install or Upgrade DM-monitor"
-      shell: "ansible-playbook {{ dm_monitor_playbook }}"
-      become: yes
-      ignore_errors: yes
-      when: dm_monitor_playbook
-
-    # Delete old Deployment Manager pod if it was reinstalled,
-    # the pod to delete may already be terminated by helm
-    - name: Restart Deployment Manager if reinstalled
-      command: >-
-        kubectl -n platform-deployment-manager delete pods {{ deployment_manager_pod_name.stdout }}
-      environment:
-        KUBECONFIG: "/etc/kubernetes/admin.conf"
-      when: deployment_manager_pod_name.stdout is defined and deployment_manager_pod_name.stdout
-      failed_when: false
+      when: not reconfig_flag
```

Comment:
May need to update the block for the following cases:

- application was uploaded - can we upload it again?
- application was upload-failed: I assume this block should be also fine in this case
- application was applying - should only waiting for applied
- application was removing:  I assume this block should be also fine in this case, suggest to double check
- application was remove-failed:  I assume this block should be also fine in this case, suggest to double check